### DISCUSSION TITLE
Devices dropping out

### DISCUSSION THREAD

**Discussion by Diramu** [make-all/tuya-local#1187] [Q&A]
I've glance through the issues and discussions and might have missed it issue, so apologies if I missed this.

I have 27 Tuya device running under tuya-local, these devices can also be controlled by Tuya App. I haven't cut them off (yet).
But occasionally, a device no  longer responses to HA, but still works fine with Tuya.  I just try fixing two devices  that did not respond. I went through the hassle for find there device_ids and local_key again, and they had not changed.

Just going to the devices  and hitting configure in  the integration  (it came up with the correct key) did not work. Deleting the device, and added a new device did work.  This came lead to new entities name which came be a hassel. 

I guess the questions I asking are
1) why are devices dropping out?
2) is there a better way to restore them?
3) Should I create an issue?

This problem does not happen very often.

**Comment by make-all** [comment]
1. isn't possible to answer with so little information, but it is possible that the WiFi signal is too weak for the printed antennas on the Tuya modules to pick up reliably, or the network overloaded.
2. Just reloading the integration or worst case restarting HA should work.  If the devices are fully configured with a static IP address and known protocol version instead of setting those to "Auto", then they will have a better chance of recovering by themselves from any network issues.
3. Only if you have a specific error that you have traced this to, otherwise the assumption would be that this is caused by external factors outside the control of this integration.

**Reply by Diramu** [reply]
Thanks for the feedback.  

My wifi network is pretty good. Unifi with 3 access points. 

All devices are added to the config with fixed ip address, but I don't recall very setting the protocol, so I guess they could be Auto.
The devices themself were not assigned static ip address, but Unifi is very good at not changing address once assigned to a specific MAC address.  But there is a chance...
I'll try an integration reload next time (or a HA restart).

Any ideas how to increase debugging, or what to look for in the logs?

**Reply by make-all** [reply]
Your Wifi network might be pretty good, but the quality of the WiFi modules used in Tuya devices aren't affected by how good your router/mesh setup is.  In particular, they only support 2.4GHz, so are affected by interference from microwave ovens and anything else on that crowded band.

The low level network communication is handled by the tinytuya library, which has a package prefix of tinytuya, so you can configure debug logging for that that to get more logs than just custom_components.tuya-local debug logs.
If you are looking for how to configure logging in general, you can find it in the [Home Assistant docs](https://www.home-assistant.io/integrations/logger/)
